reports/report_service.py

```diff
@@ -1,4 +1,32 @@
+import os
+
+from reports.data_helper import prepare_expense_data, prepare_income_data
+
+
 class ReportService:
+
     def __init__(self, db):
         self.db = db
-    
\ No newline at end of file
+
+    @staticmethod
+    def generate_report(transactions, output_dir="../static/reports/bank_of_scotland", export_format="csv"):
+        expense_df = prepare_expense_data(transactions)
+        income_df = prepare_income_data(transactions)
+
+        if export_format in ["csv", "json"]:
+            os.makedirs(output_dir, exist_ok=True)
+
+        if export_format == "csv":
+            expense_csv_path = os.path.join(output_dir, "expenses.csv")
+            income_csv_path = os.path.join(output_dir, "incomes.csv")
+            expense_df.to_csv(expense_csv_path, index=False)
+            income_df.to_csv(income_csv_path, index=False)
+            print(f"Expenses and incomes exported to {output_dir} as CSV files.")
+            return
```

Comment:
Can be splitted into function or perhaps would be nice to create `serialisation.py` file with a class implemented there

```python
class BaseSerialiser: 
     ...

class BankOfScotlandserialiser():
    ... 
```

And call the methods based on the input utilising handlers

```python

serialisation_options = {
    "csv" : BaseSerialiser.serialise_csv()
    "json": BaseSerialiser.serialise_json()
}

serialisation_options["csv"]()

Reply:
Separated as described in the PR comment, added `serialisation.py` file.